apps/docs/src/app/core/component-docs/alert/alert-docs.component.ts

```diff
@@ -128,11 +138,30 @@ export class AlertDocsComponent {
 
     schema: Schema;
 
+    @ViewChild('alert')
+    alertComponent: AlertComponent;
+
     constructor(private schemaFactory: SchemaFactoryService) {
         this.schema = this.schemaFactory.getComponent('alert');
     }
 
     onSchemaValues(data): void {
         this.data = data;
     }
+
+    /** opens alert */
+    openDynamicAlert(): void {
+        this.shouldShow = true;
+        setTimeout(() => {
```

Comment:
why the `setTimeout` ?

Reply:
It is to put time gap between between `shouldShow` becomes `true` and `open()` gets called. Otherwise, open() gets called before the `fd-alert` loads on `shouldShow=true` and results in `undefined`.